lib/active_record_upsert/active_record/persistence.rb

```diff
@@ -2,12 +2,14 @@ module ActiveRecordUpsert
   module ActiveRecord
     module PersistenceExtensions
 
-      def upsert
+      def upsert(attribute_names=nil)
         raise ::ActiveRecord::ReadOnlyRecord, "#{self.class} is marked as readonly" if readonly?
         raise ::ActiveRecord::RecordSavedError, "Can't upsert a record that has already been saved" if persisted?
         values = run_callbacks(:save) {
           run_callbacks(:create) {
-            _upsert_record
+            attribute_names ||= changed
+            attribute_names = attribute_names.map(&:to_s) + ['created_at', 'updated_at']
```

Comment:
@me Possibly by defining methods:
`timestamp_attributes_for_create` and `timestamp_attributes_for_update`. Need to check.